prepare.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python3
+'''
+Copyright © 2023 Kwankyu Lee <ekwankyu@gmail.com>
+                 Harald Schilly <harald.schilly@gmail.com>
+
+Prepare SageMath Documentation for publishing to CDN
+
+- copy `$SAGE_ROOT/local/share/doc/sage/html` to `html` directory
+- copy `$SAGE_ROOT/local/share/doc/sage/pdf` to `pdf` directory
+- create root index file from `$SAGE_ROOT/local/share/doc/sage/index.html`
+- create `robots.txt` file
+- create minimal `index.html` files in subdirectories
+- remove `_static` symbolic links and resolve the links in html files
+
+Note: the script assumes the html and pdf files were built by running doc-html
+and doc-pdf targets. e.g. like this:
+
+  $ nice ionice -c 3 sh -c 'make -j2 && make doc-html && make doc-pdf'
+
+'''
+
+import os
+import sys
+import shutil
+import configparser
+import logging
+import subprocess as sp
+
+from old_versions import OLD_VERSIONS
+
+logging.basicConfig(level="INFO", format="%(message)s")
+log = logging.getLogger(__name__)
+
+default_config = {
+    'sage_root' : '',
+    'show_older_versions' : 'false',
+}
+
+config = configparser.ConfigParser(default_config)
+if len(sys.argv) > 1 :
+    config.read(sys.argv[1])
+else :
+    config.read('prepare.cfg')
+
+sage_root = config.get('source', 'sage_root')
+
+show_older_versions = config.getboolean('target', 'show_older_versions')
+
+# ---------------------------------------------------------------
+# copy `$SAGE_ROOT/local/share/doc/sage/html` to `html` directory
+# ---------------------------------------------------------------
+
+log.info('### Copy html documentation')
+
+sp.call(f"rsync -rlp --delete {sage_root}/local/share/doc/sage/pdf ./".split(), stdout=sp.PIPE)
```

Comment:
currently I do `rm directory` + `cp hardlink` instead of rsync. I'll change that on my end.